samples/android-kotlin/app/src/main/java/com/android/virgilsecurity/ethreesamplekotlin/EThreeActivity.kt

```diff
@@ -0,0 +1,310 @@
+                    e.printStackTrace()
+                }
+
+                // After you successfully authenticated your user - you have to initialize EThree SDK.
+                // To do this you have to provide context and two listeners.
+                // OnGetTokenCallback should exchange recently received authToken for a Virgil JWT.
+                // OnResultListener<EThree> will give you initialized instance of EThree SDK in onSuccess method.
+                EThree.initialize(
+                    this@EThreeActivity,
+                    onGetTokenUserOneCallback,
+                    onInitUserOneListener
+                )
+            }
+
+            override fun onError(throwable: Throwable) {
+                throwable.printStackTrace()
+                runOnUiThread { Toast.makeText(this@EThreeActivity, throwable.message, Toast.LENGTH_SHORT).show() }
+            }
+        })
+    }
+
+
+    /**
+     * If you use some normal architecture like MVP, MVC, MVVM - you can get rid of the Callback hell. As well
+     * you can wrap all callbacks to RxJava and make it really concise. But you can use just callbacks. It's up to you.
+     */
+    fun initUserTwo() {
+        // You start your user authentication/authorization (signUp/signIn) here.
+        authenticate(identityTwo, object : OnResultListener<String> {
+            override fun onSuccess(value: String) {
+                try {
+                    val `object` = JSONObject(value)
+                    authTokenUserTwo = `object`.get("authToken") as String
+                } catch (e: JSONException) {
+                    e.printStackTrace()
+                }
+
+                EThree.initialize(
+                    this@EThreeActivity,
+                    onGetTokenUserTwoCallback,
+                    onInitUserTwoListener
+                )
+            }
+
+            override fun onError(throwable: Throwable) {
+                throwable.printStackTrace()
+                runOnUiThread { Toast.makeText(this@EThreeActivity, throwable.message, Toast.LENGTH_SHORT).show() }
+            }
+        })
+    }
+
+    /**
+     * !!! Current implementation of *authentication* is NOT SECURE and is just to show you that before accessing your
+     * service that generates Virgil JWT's you have to authenticate your user.
+     *
+     *
+     * In this method you have to use your authentication - Firebase, Google, Facebook, or any other auth (as well
+     * as some kind of your custom authentication service).
+     *
+     * @param identity         is identityOne of user for authentication
+     * @param onResultListener is a callback where you you will receive authentication token that you can later exchange
+     * for a Virgil JWT. Or error if something went wrong.
+     */
+    private fun authenticate(
+        identity: String,
+        onResultListener: OnResultListener<String>
+    ) {
+        try {
+            val baseUrl = "http://10.0.2.2:3000/authenticate"
```

Comment:
Let's move it to BuildConfig

Reply:
This demo should be stupidly simple, so I think it's better to leave all info in this class